src/guards/roles.guard.ts

```diff
@@ -0,0 +1,33 @@
+import {
+    Injectable,
+    CanActivate,
+    ExecutionContext,
+    Inject,
+} from '@nestjs/common';
+import { Reflector } from '@nestjs/core';
+import { UserService } from '../user/user.service';
+import { User } from 'src/model/user.model';
+
+@Injectable()
+export class RolesGuard implements CanActivate {
+    constructor(
+        @Inject('UserService') private readonly userService: UserService,
+        private readonly reflector: Reflector,
+    ) {}
+
+    async canActivate(context: ExecutionContext): Promise<boolean> {
+        const roles = this.reflector.get<string[]>(
+            'roles',
+            context.getHandler(),
```

Comment:
In order to use `@RolesGuard`, `context.getHandler()` should be `context.getClass()` instead.
reference: https://docs.nestjs.com/fundamentals/execution-context

Reply:
Thanks